gemonlinedb/include/gem/onlinedb/DBInterface.h

```diff
@@ -0,0 +1,54 @@
+#ifndef GEM_ONLINEDB_DBINTERFACE_H
+#define GEM_ONLINEDB_DBINTERFACE_H
+
+#include <xdata/Table.h>
+
+#include "gem/onlinedb/detail/RegisterData.h"
+
+namespace gem {
+    namespace onlinedb {
+        /**
+         * @brief Used to interface with the database.
+         */
+        class DBInterface
+        {
+        public:
+            /**
+             * @brief Creates a configuration object from a database row.
+             */
+            template<class ConfigurationType>
+            static ConfigurationType convertRow(const xdata::Table &table,
+                                                xdata::Table::Row &row);
+        };
+
+        template<class ConfigurationType>
+        ConfigurationType DBInterface::convertRow(const xdata::Table &table,
+                                                  xdata::Table::Row &row)
+        {
+            auto columns = table.getColumns();
+            detail::RegisterData data;
+            for (const auto &c : columns) {
+                auto value = row.getField(c);
+                if (value != nullptr && value->type() == "string") {
+                    auto str = value->toString();
+                    if (!str.empty()) {
+                        data[c] = std::stoi(value->toString());
+                    } else {
+                        // Empty value in database -> use 0
+                        // This *does* actually happen...
```

Comment:
An exception has to be thrown such that caller is notified about DB incompleteness

Reply:
Unfortunately it would also prevent me from testing with the existing DB. In addition if new fields are added later they will be initialized to `null` (and would hence break existing data).

Reply:
Can't you introduce catchers in your test code? The throwing an exception does not always mean an error as it can be intercepted. In the current design a user won't know about flaw in the DB unless explicitly check the values (and 0 could be actually in valid range!)

Reply:
Since only the AMC configuration is affected I disabled the corresponding test.